File: app/src/main/java/org/mifos/mobilebanking/utils/ProgressBarHandler.java
package org.mifos.mobilebanking.utils;

import android.R.attr;
import android.R.id;
import android.app.Activity;
import android.content.Context;
import android.view.Gravity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ProgressBar;
import android.widget.RelativeLayout;
import android.widget.RelativeLayout.LayoutParams;


/**
 * @author Rajan Maurya
 */
public class ProgressBarHandler {

    private ProgressBar mProgressBar;
    private Context mContext;

    public ProgressBarHandler(Context context) {
        mContext = context;

        ViewGroup layout = (ViewGroup) ((Activity) context).findViewById(id.content).getRootView();

        mProgressBar = new ProgressBar(context, null, attr.progressBarStyleInverse);
        mProgressBar.setIndeterminate(true);

        LayoutParams params = new
                LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT);

        RelativeLayout rl = new RelativeLayout(context);

        rl.setGravity(Gravity.CENTER);
        rl.addView(mProgressBar);

        layout.addView(rl, params);

        hide();
    }

    public void show() {
        mProgressBar.setVisibility(View.VISIBLE);
    }

    public void hide() {
        mProgressBar.setVisibility(View.INVISIBLE);
    }
}

File: app/src/main/java/org/mifos/mobile/utils/ProgressBarHandler.kt
package org.mifos.mobile.utils

import android.R.attr
import android.R.id
import android.app.Activity
import android.content.Context
import android.view.Gravity
import android.view.View
import android.view.ViewGroup
import android.widget.ProgressBar
import android.widget.RelativeLayout

/**
 * @author Rajan Maurya
 */
class ProgressBarHandler(private val mContext: Context) {
    private val mProgressBar: ProgressBar
    fun show() {
        mProgressBar.visibility = View.VISIBLE
    }

    fun hide() {
        mProgressBar.visibility = View.INVISIBLE
    }

    init {
        val layout = (mContext as Activity).findViewById<View>(id.content).rootView as ViewGroup
        mProgressBar = ProgressBar(mContext, null, attr.progressBarStyleInverse)
        mProgressBar.isIndeterminate = true
        val params = RelativeLayout.LayoutParams(RelativeLayout.LayoutParams.MATCH_PARENT, RelativeLayout.LayoutParams.MATCH_PARENT)
        val rl = RelativeLayout(mContext)
        rl.gravity = Gravity.CENTER
        rl.addView(mProgressBar)
        layout.addView(rl, params)
        hide()
    }
}